assets/js/job-dashboard.js

```diff
@@ -1,8 +1,49 @@
 /* global job_manager_job_dashboard */
-jQuery(document).ready(function($) {
 
-	$('.job-dashboard-action-delete').click(function() {
-		return window.confirm( job_manager_job_dashboard.i18n_confirm_delete );
-	});
+import domReady from '@wordpress/dom-ready';
 
-});
\ No newline at end of file
+// eslint-disable-next-line camelcase
+const { i18nConfirmDelete, overlayEndpoint } = job_manager_job_dashboard;
+
+function setupEvents( root ) {
+	root
+		.querySelectorAll( '.job-dashboard-action-delete' )
+		.forEach( el => el.addEventListener( 'click', confirmDelete ) );
+}
+
+function confirmDelete( event ) {
+	// eslint-disable-next-line no-alert
+	if ( ! window.confirm( i18nConfirmDelete ) ) {
+		event.preventDefault();
+	}
+}
+
+async function showOverlay( event ) {
+	const overlayDialog = document.getElementById( 'jmDashboardOverlay' );
+
+	if ( ! overlayDialog ) {
+		return true;
+	}
+
+	event.preventDefault();
+	overlayDialog.showModal();
+
+	const contentElement = overlayDialog.querySelector( '.jm-dialog-modal-content' );
+	contentElement.innerHTML = '<a class="jm-ui-spinner"></a>';
+
+	const { success, data } = await (
+		await fetch( `${ overlayEndpoint }?job_id=${ this.dataset.jobId }` )
+	 ).json();
```

Comment:
What if this throws an error? (e.g. a connection error or something)

Can we have some error treatment for that case here?

Reply:
Good idea, added some error handling in a672d90